### PR TITLE
Use foxbox/users

### PR DESCRIPTION
**Pull request by ferjm**
r? @fabricedesre

### CODE REVIEW

**Comment by jedireza** [comment]
I don't think this needs to be an external crate. It adds a level of indirection that I don't think is necessary.

Could this approach cause version drift across `foxbox` and `users` code? Does this add complexity to maintenance?

**Comment by ferjm** [comment]
Cargo should handle the versioning. If something changes in fxbox/users, your fxbox/foxbox clone won't get the changes until you make "cargo update" locally. If that's too much trouble I can merge fxbox/users here.

File: src/http_server.rs [hunk 1]
```diff
@@ -2,7 +2,10 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
+extern crate foxbox_users;
```
Comment (fabricedesre) [inline src/http_server.rs]: Any reason to not move that in main.rs with the other external crates?

File: src/http_server.rs [hunk 2]
```diff
@@ -56,9 +59,12 @@ impl HttpServer {
             }
         });
 
+        let users_router = UsersRouter::new();
+
         let mut mount = Mount::new();
         mount.mount("/", Static::new(Path::new("static")))
-             .mount("/services", router);
+             .mount("/services", router)
+             .mount("/users_admin", users_router);
```
Comment (fabricedesre) [inline src/http_server.rs]: Can you inline UsersRouter::new() here?

Also, maybe /users as a url?

**Comment by fabricedesre** [comment]
I'd also like to keep that as an external crate. This is one piece that could be useful to others, along some other like the mDNS & openzwave libraries.

File: src/http_server.rs (hunk 2 above)
Comment (ferjm) [inline src/http_server.rs]: There are several [endpoints ](https://github.com/fxbox/users/blob/master/doc/API.md#api-endpoints) with a path already including `users` so I figured that `users_admin` would be a better fit. So we would have urls like: https://foxbox.local/users_admin/setup https://foxbox.local/users_admin/users etc.

Comment (fabricedesre) [inline src/http_server.rs]: ok!

**Pull request merged**